### PR TITLE
[17.0][ADD] sale_order_line_auto_section: order line automatic sections by product category

### PR DESCRIPTION
**Pull request by dreispt**

### CODE REVIEW

File: sale_order_line_auto_section/models/product_category.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2025 Open Source Integrators
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import fields, models
+
+
+class ProductCategory(models.Model):
+    _inherit = "product.category"
+    _order = "section_sequence,name"
```
Comment (alexey-pelykh) [inline sale_order_line_auto_section/models/product_category.py:9]: Changing `_order` on `product.category` is a global side effect that changes ordering throughout the entire Odoo instance. The sorting is already handled in `action_organize_lines_by_category` via `sorted()` and the list view has `default_order`. Please remove this `_order` override.

File: sale_order_line_auto_section/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            for j, line in enumerate(section_lines, 1):
+                line.sequence = section.sequence + j
+
+        # Move manual sections/notes to the end
+        manual_lines = self.order_line.filtered(
+            lambda line: line.display_type in ["line_section", "line_note"]
+            and not line.auto_section_category_id
+        )
+        if manual_lines:
+            max_sequence = max(self.order_line.mapped("sequence") or [100])
```
Comment (alexey-pelykh) [inline sale_order_line_auto_section/models/sale_order.py:93]: Fallback `or 9` is inconsistent with the field default of `10`. Since `section_sequence` is an Integer with `default=10`, the `or 9` only triggers when `sequence == 0`, sorting it before defaults. Remove the fallback.

File: sale_order_line_auto_section/README.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+============================
+
+.. 
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   !! This file is generated by oca-gen-addon-readme !!
+   !! changes will be overwritten.                   !!
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   !! source digest: sha256:placeholder
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
```
Comment (alexey-pelykh) [inline sale_order_line_auto_section/README.rst:12]: Source digest is `sha256:placeholder` -- needs regeneration with `oca-gen-addon-readme`.

**Review by alexey-pelykh — COMMENTED** [review]
Thanks for this contribution -- auto-sectioning sale order lines by product category is a genuinely useful feature and the implementation is solid. Tests are comprehensive (8 methods covering flow, sequencing, idempotency, edge cases).

One significant concern and a few minor items:

**`_order` override on `product.category` is too invasive.** Changing `_order` to `"section_sequence,name"` affects every product category list/search across the entire instance (Accounting, Inventory, Purchase, etc.) -- not just in the sale order context. The sorting is already handled correctly in `action_organize_lines_by_category` via `sorted()`, and the list view has `default_order`. Please remove the `_order` override.

**Sort key fallback `or 9` is inconsistent** with the field default of `10`. Since `section_sequence` is an Integer with `default=10`, the `or 9` branch only triggers when `sequence == 0`, placing it before default categories. Remove the fallback.

**Consider adding `groups="sales_team.group_sale_salesman"`** to the button for consistency with other sale order action buttons.

**README source digest is `sha256:placeholder`** -- needs regeneration with `oca-gen-addon-readme`.

File: sale_order_line_auto_section/models/product_category.py (hunk 1 above)
Comment (rousseldenis) [reply sale_order_line_auto_section/models/product_category.py:9]: @dreispt Indeed, is this a really needeed change as can have an impact on whole stack ?
